docs: Update Deep Links analysis document (#1443)
Flank/flank-dashboard: metrics/web/docs/features/deep_links/01_deep_links_analysis.md

> # Deep Links Design

Reviewer: ```suggestion
# Deep Linking
```
Also, does it make sense to rename the file to be `01_deep_linking.md`?

> Feature description / User story.

As a user, I want to pass direct links to project and/or project groups, so that other users can navigate easier.

# Analysis
> Describe the general analysis approach.

The `Deep Linking` feature improves the navigation experience by allowing users to navigate to a specific resource of an application and to save the application state in a URL.

The analysis stage aims the following goals:
1. Define the detailed requirements for the deep linking feature;

> 2. Investigate existing implementation approaches for the deep linking in the Flutter Web application.

Reviewer: ```suggestion
2. Investigate existing approaches for the deep linking in the Flutter Web applications.
```
?

> 3. Select the implementation approach for the Metrics Web application that will satisfy the requirements.

Reviewer: I believe we are not choosing the implementation approach during the analysis. The analysis goal is to investigate any available solutions and understanding if they are suitable for us or not. So, we should overview any existing solutions here (if any) and choose whether we are going to use the existing one, or develop our own approach suitable for our requirements.

Cons:
- May require code generation;
- Requires the navigation system reintegration - meaning that we must use the 3-rd party `RouteInformationParser`s and `RouterDelegate`s.

## Custom implementation
Consider the following pros and cons of the custom implementation approach:

Pros:
- Highly customizable approach;
- Well-testable approach;

Cons:
- Requires more code comparing to the 3-rd party package approach.

According to the comparison above, we choose the [`Custom implementation`](#custom-implementation) approach, as it fits the current navigation approach used in the Metrics Web application, and satisfies the customization and testability requirements.

Consider this diagram that briefly describes the custom implementation approach:

![Custom approach diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://raw.githubusercontent.com/Flank/flank-dashboard/update_deep_links_analysis/metrics/web/docs/features/deep_links/diagrams/custom_approach_component_diagram.puml)

# Prototyping
> Create a simple prototype to confirm that implementing this feature is possible.

> The following code snippet demonstrates the process of parsing the route parameters from the URL:

Reviewer: ```suggestion
The following code snippet demonstrates the process of parsing the query parameters from the URL:
```
?
